api/segments/serializers.py

```diff
@@ -74,13 +85,38 @@ def validate_project_segment_limit(self, project: Project) -> None:
                 }
             )
 
-    def update(self, instance, validated_data):
-        # use the initial data since we need the ids included to determine which to update & which to create
-        rules_data = self.initial_data.pop("rules", [])
-        self._update_segment_rules(rules_data, segment=instance)
-        # remove rules from validated data to prevent error trying to create segment with nested rules
-        del validated_data["rules"]
-        return super().update(instance, validated_data)
+    def validate_segment_rules_conditions_limit(
+        self, rules_data: dict[str, object]
+    ) -> None:
+        count = self._calculate_condition_count(rules_data)
+
+        if self.instance and getattr(self.instance, "whitelisted_segment", None):
+            return
```

Comment:
We could probably swap these 2 statements around, right? If it is a whitelisted segment, then it must exceed the limit.

Reply:
Oh yeah that was dumb of me. Will move the early return.